lms/djangoapps/teams/static/teams/js/views/edit_team.js

```diff
@@ -0,0 +1,168 @@
+                   var teamDescription = this.teamDescriptionField.fieldValue();
+                   var teamLanguage = this.teamLanguageField.fieldValue();
+                   var teamCountry = this.teamCountryField.fieldValue();
+
+                   var validation = this.validateTeamData(teamName, teamDescription);
+                   if (validation.status === false) {
+                       this.showMessage(validation.message);
+                       return;
+                   }
+
+                   var data = {
+                       course_id: this.courseId,
+                       topic_id: this.topicId,
+                       name: teamName,
+                       description: teamDescription,
+                       language: _.isNull(teamLanguage) ? '' : teamLanguage,
+                       country: _.isNull(teamCountry) ? '' : teamCountry
+                   };
+
+                   // Send AJAX request to Teams API
+                   var view = this;
+                   $.ajax({
+                       type: 'POST',
+                       url: this.teamsUrl,
+                       data: data
+                   }).done(function () {
+                       Backbone.history.navigate("topics/" + view.topicId, {trigger: true});
+                   }).fail(function (jqXHR) {
+                       view.showMessage(gettext('An error occurred. Please try again.'));
+                   });
+               },
+
+               validateTeamData: function (teamName, teamDescription) {
+                   var status = true,
+                       message = gettext("Your team could not be created. Check the highlighted fields below and try again.");
+
+                   this.teamNameField.unhighlightField();
+                   this.teamDescriptionField.unhighlightField();
+
+                   if (_.isEmpty(teamName.trim()) ) {
+                       status = false;
+                       this.teamNameField.highlightFieldOnError();
+                   } else if (teamName.length > this.maxTeamNameLength) {
+                       status = false;
+                       this.teamNameField.highlightFieldOnError();
+                   }
+
+                   if (_.isEmpty(teamDescription.trim()) ) {
+                       status = false;
+                       this.teamDescriptionField.highlightFieldOnError();
+                   } else if (teamDescription.length > this.maxTeamDescriptionLength) {
+                       status = false;
+                       this.teamDescriptionField.highlightFieldOnError();
+                   }
+
+                   return {
+                       status: status,
+                       message: message
+                   };
+               },
+
+               showMessage: function (message) {
+                   this.$('.wrapper-msg').show();
+                   this.$('.msg-content .copy p').text(message);
+               },
+
+               cancelTeam: function () {
```

Comment:
You can rename this `goBackToTopic()` and also use in the ajax request's `done()` callback.